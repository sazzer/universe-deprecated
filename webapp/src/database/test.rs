use lazy_static::lazy_static;
use log::info;
use postgres::{Client, NoTls};
use testcontainers::{clients::Cli, images::postgres::Postgres, Container, Docker};

lazy_static! {
    static ref DOCKER: Cli = { Cli::default() };
}

pub struct TestDatabase<'d> {
    #[allow(dead_code)]
    node: Container<'d, Cli, Postgres>,
    url: String,
}

impl<'d> TestDatabase<'d> {
    pub fn new() -> Self {
        let node = DOCKER.run(Postgres::default());

        let host_port = node.get_host_port(5432).unwrap();
        let url = format!("postgres://postgres:postgres@localhost:{}", host_port);
        info!("Running postgres on {}", url);

        TestDatabase { node, url }
    }

    pub fn client(&self) -> Client {
        Client::connect(&self.url, NoTls).unwrap()
    }
}
