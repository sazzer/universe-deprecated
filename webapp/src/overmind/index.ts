import { namespaced } from "overmind/config";
import { createHook } from "overmind-react";
import { IConfig } from "overmind";
import * as login from "../login";
import * as authentication from "../authentication";

export const config = namespaced({
  login,
  authentication
});

export const useOvermind = createHook<typeof config>();

declare module "overmind" {
  // tslint:disable:interface-name
  interface Config extends IConfig<typeof config> {}
}
