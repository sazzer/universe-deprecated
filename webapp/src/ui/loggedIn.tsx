import React from "react";
import { useTranslation } from "react-i18next";
import { useUser } from "../users";

/**
 * Wrapper around other components to only render if there is a currently logged in user
 */
export const LoggedIn: React.FC = ({ children }) => {
  const { hasUser } = useUser();
  const { t } = useTranslation();

  const body = hasUser ? children : t("errors.unauthenticated");

  return <>{body}</>;
};
