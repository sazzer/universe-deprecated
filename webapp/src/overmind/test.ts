import { createOvermindMock, OvermindMock, Config } from "overmind";
import { config } from "../overmind";
import { cloneDeep, merge } from "lodash";

/**
 * NestedPartial is stolen from Overmind to represent a part of the Overmind structure
 */
export declare type NestedPartial<T> = T extends Function
  ? T
  : Partial<
      {
        [P in keyof T]: NestedPartial<T[P]>;
      }
    >;

/**
 * Create a test instance of the Overmind to work with
 * @param mockEffects The mock effects to wire in
 * @param initialState The initial state to wire in
 */
export function createTestOvermind(
  mockEffects?: NestedPartial<Config["effects"]>,
  initialState?: NestedPartial<Config["state"]>
): OvermindMock<Config> {
  const mergedState = merge(cloneDeep(config), { state: initialState });
  return createOvermindMock(mergedState, mockEffects);
}
