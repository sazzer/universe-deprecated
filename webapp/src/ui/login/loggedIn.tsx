import React from "react";
import { useOvermind } from "../../overmind";

/**
 * React Component to render the children only when there is an active logged-in session
 */
export const LoggedIn: React.FC = ({ children }) => {
  const { state } = useOvermind();

  if (state.authentication.isLoggedIn) {
    return <>{children}</>;
  } else {
    return <div>You must be logged in for this</div>;
  }
};
