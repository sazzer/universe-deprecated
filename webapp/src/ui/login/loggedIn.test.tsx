import React from "react";
import { render } from "@testing-library/react";
import { LoggedIn } from "./loggedIn";
import { createTestOvermind } from "../../overmind/test";
import { Provider } from "overmind-react";
import { OvermindMock, Config } from "overmind";

describe("When not logged in", () => {
  test("It renders correctly", () => {
    const overmind = createTestOvermind();
    const { container } = render(
      <Provider value={overmind}>
        <LoggedIn>Hello</LoggedIn>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});

describe("When logged in", () => {
  test("It renders correctly", () => {
    const overmind = createTestOvermind(
      {},
      {
        users: {
          users: {
            someUserId: {
              userId: "someUserId",
              username: "username",
              email: "test@example.com",
              displayName: "Test User"
            }
          }
        },
        authentication: {
          userId: "someUserId",
          accessToken: {
            accessToken: "accessToken",
            expires: new Date().toISOString()
          },
          isLoggedIn: true
        }
      }
    );
    const { container } = render(
      <Provider value={overmind}>
        <LoggedIn>Hello</LoggedIn>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
