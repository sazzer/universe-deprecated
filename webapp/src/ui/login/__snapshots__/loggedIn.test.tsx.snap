// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When logged in It renders correctly 1`] = `
<div>
  Hello
</div>
`;

exports[`When not logged in It renders correctly 1`] = `
<div>
  <div>
    You must be logged in for this
  </div>
</div>
`;
